src/pages/Dashboard/Modals/SupplyWithdraw/index.tsx

```diff
@@ -0,0 +1,160 @@
+/** @jsxImportSource @emotion/react */
+import React, { useState } from 'react';
+import BigNumber from 'bignumber.js';
+import {
+  ConnectWallet,
+  EnableToken,
+  Tabs,
+  Modal,
+  IModalProps,
+  Token,
+  IInfoRowProps,
+} from 'components';
+import { AmountForm, IAmountFormProps } from 'containers/AmountForm';
+import { useTranslation } from 'translation';
+import { Asset, TokenId } from 'types';
+import { formatApy } from 'utilities/common';
+import SupplyWithdrawForm from './SupplyWithdrawForm';
+import { useStyles } from '../styles';
+
+export interface ISupplyWithdrawUiProps {
+  className?: string;
+  onClose: IModalProps['handleClose'];
+  asset: Asset | undefined;
+  userTotalBorrowBalance: BigNumber;
+  userTotalBorrowLimit: BigNumber;
+  dailyEarnings: BigNumber;
+}
+
+const TAB_TEXT_KEYS = [
+  {
+    connect: 'supplyWithdraw.connectWalletToSupply',
+    enable: 'supplyWithdraw.enableToSupply',
+  },
+  {
+    connect: 'supplyWithdraw.connectWalletToWithdraw',
+    enable: 'supplyWithdraw.enableToWithdraw',
+  },
+];
+
+/**
+ * The fade effect on this component results in that it is still rendered after the asset has been set to undefined
+ * when closing the modal.
+ */
+export const SupplyWithdrawUi: React.FC<ISupplyWithdrawUiProps> = ({
+  className,
+  onClose,
+  asset,
+  userTotalBorrowBalance,
+  userTotalBorrowLimit,
+  dailyEarnings,
+}) => {
+  const styles = useStyles();
+  const [activeTabIndex, setActiveTabIndex] = useState<number>(0);
+
+  const { t, Trans } = useTranslation();
+  const { id: assetId, isEnabled } = asset || {};
+  const activeTabTextKeys = TAB_TEXT_KEYS[activeTabIndex];
+  const tokenInfo: IInfoRowProps['info'] = asset
+    ? [
+        {
+          left: t('supplyWithdraw.supplyApy'),
+          symbol: assetId as TokenId,
+          right: formatApy(asset.supplyApy),
+        },
+        {
+          left: t('supplyWithdraw.distributionApy'),
+          symbol: 'xvs',
+          right: formatApy(asset.xvsSupplyApy),
+        },
+      ]
+    : [];
+
+  const onSubmit: IAmountFormProps['onSubmit'] = () => {
+    // TODO: https://app.clickup.com/t/24quhp4
+  };
+  const newSupplyAmount = (amount: BigNumber) => userTotalBorrowLimit.plus(amount);
+  const newBorrowAmount = (amount: BigNumber) => userTotalBorrowLimit.minus(amount);
```

Comment:
I can rename them so they are clearer. This are required because depending on which form is rendered we will add or subtract the user input amount from `userTotalBorrowLimit`